examples/edwards25519-hash/src/edwards25519-hash.rs

```diff
@@ -0,0 +1,534 @@
+) -> ByteSeq {
+    let ell = (len_in_bytes + B_IN_BYTES - 1) / B_IN_BYTES; // ceil(len_in_bytes / b_in_bytes)
+                                                            // must be that ell <= 255
+    let dst_prime = dst.push(&U8_from_usize(dst.len())); // DST || I2OSP(len(DST), 1)
+    let z_pad = ByteSeq::new(S_IN_BYTES); // I2OSP(0, s_in_bytes)
+    let mut l_i_b_str = ByteSeq::new(2);
+    l_i_b_str[0] = U8_from_usize(len_in_bytes / 256);
+    l_i_b_str[1] = U8_from_usize(len_in_bytes); // I2OSP(len_in_bytes, 2)
+
+    let msg_prime = z_pad
+        .concat(msg)
+        .concat(&l_i_b_str)
+        .concat(&ByteSeq::new(1))
+        .concat(&dst_prime); // Z_pad || msg || l_i_b_str || 0 || dst_prime
+    
+    let b_0 = ByteSeq::from_seq(&hash(&msg_prime)); // H(msg_prime)
+    let mut b_i = ByteSeq::from_seq(&hash(&b_0.push(&U8(1u8))
+        .concat(&dst_prime))); // H(b_0 || 1 || dst_prime)
+    let mut uniform_bytes = ByteSeq::from_seq(&b_i);
+
+    for i in 2..(ell + 1) {
+        let t = ByteSeq::from_seq(&b_0);
+        b_i = ByteSeq::from_seq(&hash(&(t ^ b_i).push(&U8_from_usize(i))
+            .concat(&dst_prime))); //H((b_0 ^ b_(i-1)) || 1 || dst_prime)
+        uniform_bytes = uniform_bytes.concat(&b_i);
+    }
+    uniform_bytes.truncate(len_in_bytes)
+}
+
+// Adapted from bls12-381-hash.rs
+// Specification can be seen in section 5.2 of https://datatracker.ietf.org/doc/draft-irtf-cfrg-hash-to-curve/
+pub fn ed_hash_to_field(
+    msg: &ByteSeq, dst: &ByteSeq, count: usize
+) -> Seq<Ed25519FieldElement> {
+    let len_in_bytes = count * L; // count * m * L
+    let uniform_bytes = expand_message_xmd(msg, dst, len_in_bytes);
+    let mut output = Seq::<Ed25519FieldElement>::new(count);
+
+    for i in 0..count {
+        // m = 1, so no loop
+        let elm_offset = L * i; // L * (j + i * m)
+        let tv = uniform_bytes.slice(elm_offset, L); //substr(uniform_bytes, elm_offset, L)
+        let u_i = Ed25519FieldElement::from_byte_seq_be(
+            &EdFieldHash::from_byte_seq_be(&tv).to_byte_seq_be().slice(32,32)); // OS2IP(tv) mod p
+        output[i] = u_i;
+    }
+    output
+}
+
+// Adapted from bls12-381-hash.rs
+fn ed_is_square(x: Ed25519FieldElement) -> bool {
+    let c1 = Ed25519FieldElement::from_byte_seq_be(&P_1_2.to_be_bytes()); // (p - 1) / 2
+    let tv = x.pow_self(c1);
+    tv == Ed25519FieldElement::ZERO() || tv == Ed25519FieldElement::ONE()
+}
+
+// Specification can be seen in section 4.1 of https://datatracker.ietf.org/doc/draft-irtf-cfrg-hash-to-curve/
+fn sgn0_m_eq_1(x: Ed25519FieldElement) -> bool {
+    x % Ed25519FieldElement::TWO() == Ed25519FieldElement::ONE()
+}
+
+// Adapted from bls12-381-hash.rs
+pub fn ed_clear_cofactor(x: EdPoint) -> EdPoint {
+    point_mul_by_cofactor(x)
+}
+
+fn cmov(
+    a: Ed25519FieldElement, b: Ed25519FieldElement, c: bool
+) -> Ed25519FieldElement {
+    if c {
+        b
+    } else {
+        a
+    }
+}
+
+fn xor(a: bool, b: bool) -> bool {
```

Comment:
This seems unnecessary. You can just do `a ^ b`

Reply:
This causes the hacspec typechecker to fail with the following error:
```
error[Hacspec]: operation not available for type bool
   --> examples/edwards25519-hash/src/edwards25519-hash.rs:244:31
    |
244 |     let y = cmov(y, zero - y, e2 ^ e3);
    |  
```
Should I open an issue regarding this?

Reply:
I filed something